File: _includes/code/python/config-refs.datatypes.uuid.py
import weaviate

# START ConfigureDataType
from weaviate.classes.config import Property, DataType
from weaviate.util import generate_uuid5

# END ConfigureDataType

client = weaviate.connect_to_local()

# Delete collection if it exists
client.collections.delete(name="Movie")

# START ConfigureDataType
# Create collection
my_collection = client.collections.create(
    name="Movie",
    properties=[
        Property(name="title", data_type=DataType.TEXT),
        Property(name="movie_uuid", data_type=DataType.UUID),
        Property(name="related_movie_uuids", data_type=DataType.UUID_ARRAY),
    ],
    # Other properties are omitted for brevity
)
# END ConfigureDataType

# START AddObject
# Create an object
example_object = {
    "title": "The Matrix",
    "movie_uuid": generate_uuid5("The Matrix"),
    "related_movie_uuids": [
        generate_uuid5("The Matrix Reloaded"),
        generate_uuid5("The Matrix Revolutions"),
        generate_uuid5("Matrix Resurrections"),
    ],
}

obj_uuid = my_collection.data.insert(example_object)
# END AddObject

returned_obj = my_collection.query.fetch_object_by_id(obj_uuid)

assert obj_uuid is not None
assert [str(i) for i in returned_obj.properties["related_movie_uuids"]] == example_object["related_movie_uuids"]

client.close()

File: _includes/code/typescript/config-refs.datatypes.uuid.ts
import assert from 'assert';
import weaviate, { WeaviateClient } from 'weaviate-client';

// START ConfigureDataType
import { dataType } from 'weaviate-client';
import { generateUuid5 } from 'weaviate-client';

// END ConfigureDataType

const client: WeaviateClient = await weaviate.connectToLocal();

await client.collections.delete('Movie');

// START ConfigureDataType
// Create collection
const myCollection = await client.collections.create({
  name: 'Movie',
  properties: [
    {
      name: 'title',
      dataType: dataType.TEXT,
    },
    {
      name: 'movie_uuid',
      dataType: dataType.UUID,
    },
    {
      name: 'related_movie_uuids',
      dataType: dataType.UUID_ARRAY,
    },
  ],
  // Other properties are omitted for brevity
});
// END ConfigureDataType

// START AddObject
const exampleObject = {
  title: 'The Matrix',
  movie_uuid: generateUuid5('The Matrix'),
  related_movie_uuids: [
    generateUuid5('The Matrix Reloaded'),
    generateUuid5('The Matrix Revolutions'),
    generateUuid5('The Matrix Resurrections'),
  ],
};

const obj_uuid = await myCollection.data.insert(exampleObject);
// END AddObject

const returnedObject = await myCollection.query.fetchObjectById(obj_uuid);

assert.notEqual(obj_uuid, null);
assert.deepEqual(returnedObject?.properties, exampleObject);

client.close();
